File: ios/Plugin/UdidPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(UdidPlugin)
public class UdidPlugin: CAPPlugin {
    private let implementation = Udid()

   @objc func getUdid(_ call: CAPPluginCall) {
        let udid = implementation.getUniqueId()
        if udid.isEmpty {
            call.reject("UDID not available")
        } else {
            call.resolve([
                "value": udid
            ])
        }
    }
}

File: android/src/main/java/com/udid/plugin/UdidPlugin.java
package com.udid.plugin;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Udid")
public class UdidPlugin extends Plugin {

    private Udid implementation;

    @Override
    public void load() {
        implementation = new Udid(getContext());
    }

    @PluginMethod
    public void getUdid(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("value", implementation.getUniqueId());
        call.resolve(ret);
    }
}
